docs/generated-files.md

```diff
@@ -0,0 +1,55 @@
+
+| File Name | Purpose | Notes |
+| :--- | :--- | :--- |
+| `xtrabackup_info` | Metadata about the backup, including the Xtrabackup version, start time, and server's MySQL version. | The file is essential for all backups. |
+| `xtrabackup_binlog_info` | Stores the binary log coordinates (position and file name) at the exact moment the backup completed. | The file is critical for point-in-time recovery. |
+| `xtrabackup_checkpoints` | Contains key checkpoint information that Xtrabackup uses to ensure a consistent restore. | The file is essential for all backups. |
+| `xtrabackup_logfile` | A copy of the InnoDB transaction log file from the server. | The log file is crucial during the prepare stage to roll back uncommitted transactions and apply committed ones. |
+| `backup-my.cnf` | A snapshot of the server's configuration file at the time of the backup. | The file is useful for understanding the server's state when the backup was taken. |
+| `xtrabackup_gtid_info` | The GTID set at the time of backup. | This file is only present if Global Transaction Identifiers (GTIDs) are enabled on the server. |
+| `xtrabackup_slave_info` | Replication metadata, including the master's binary log file and position. | This file is only created if the server is a replica. The file is used to set up the restored server as a new replica. |
+| `xtrabackup_tablespaces` | Stores metadata about the tablespaces available at the time of the backup. | The file is essential for all backups. |
+| `ib_buffer_pool` | A dump of the InnoDB buffer pool. | This optional file is created if buffer pool dumping is enabled (`innodb_buffer_pool_dump_at_shutdown`). The file is not used for the restore operation, but the file can be used to warm up the buffer pool of the restored instance, improving performance. |
+
+## Database subdirectories
+
+In addition to the files in the root directory, Xtrabackup creates subdirectories for each database present at the time of the backup. These directories contain the actual table data.
+
+* `*.ibd` files for InnoDB tables: These files store the table data and indexes. Their presence is dependent on the `innodb_file_per_table` setting being enabled in the MySQL configuration.
+
+* `*.sdi` files for non-InnoDB tables: These contain serialized data dictionary information for tables.
+
+## Incremental Backups
+
+When you perform an incremental backup, two additional files are created in the backup directory. These files are used in the apply-log phase of the restore process to merge the incremental changes into a full backup.
+
+* `delta`: This file contains the actual data changes from the InnoDB tables since the last full or incremental backup.
+
+* `meta`: This file stores metadata about the incremental backup, including the LSN (Log Sequence Number) of the last backup. This information is crucial for Xtrabackup to know where to start applying changes.
```

Comment:
meta files contain information about tablespace. No LSN.

cat t11.ibd.meta 
page_size = 8192
zip_size = 8192
space_id = 14
space_flags = 16425


During the prepare phase, the information in the .meta file is used to apply the changes in the .delta file to an InnoDB tablespace. 

Extra info:
After all the .delta files are processed, the redo log will be applied